### DISCUSSION TITLE
How to continuously print child process stdout from `.nu` script?

### DISCUSSION THREAD

**Discussion by 0x009922** [nushell/nushell#5353] [Q&A] [answered]
Hi!

I have a nu script that starts long-running child process:

```nu
# script.nu
./some_binary
```

This binary is e.g. a web server. I don't want to do anything with its logs, I just want to see them on-demand. How to achieve it?

Intuitively I tried to to `./some_binary | echo` and `./some_binary | print`.

**Comment by fdncred** [comment]
I tried to reproduce this issue with [http](https://github.com/thecoshman/http) and it seems to work fine. I'm wondering what web server you're running/testing.

<img width="1223" alt="Screen Shot 2022-04-28 at 6 49 26 AM" src="https://user-images.githubusercontent.com/343840/165745778-48febe78-28d8-4c68-ac34-100a0ece60c7.png">

```
> cat host.nu
http
```

**Reply by 0x009922** [reply]
I am running Rust binaries that are built on top of `tracing`. [Here](https://github.com/0x009922/explorer-deploy-dev-tool/blob/1f10149750bd398ff0b60ab658df07d6bb0eadba/help.nu#L43-L49) is the script itself.

Let me built a small repro-project.

**Reply by 0x009922** [reply]
Hm... it's not easy to reproduce, but not only I've stuck with it. I will come later with details.

**Comment by fdncred** [comment]
ok, sounds good. yes, a small repro would be very helpful.

**Reply by 0x009922** [reply]
Ok, I've created a minimal repro. It's weird!

TL;DR

```nu
def main [] {
  if false {
  } else if true {
    # you will not see it's stdout
    ./whatever
  }
}
```

[Repo](https://github.com/0x009922/nu-issue-eaten-stdout/tree/master)

I guess it's an issue, right?

**Reply by 0x009922** [reply]
Also, while experimenting, I've found another issue: #5362.

**Reply by 0x009922** [reply]
And here the issue for this discussion: https://github.com/nushell/nushell/issues/5363

**Comment by fdncred** [comment]
I tried to run your script and don't see 'https://github.com/soramitsu/iroha2-block-explorer-backend.git' or either it's private or something.

**Reply by 0x009922** [reply]
Yeah, it's private, but iroha is not. Logs are eaten for both.

**Comment by 0x009922** [comment, marked as answer]
Fixed: https://github.com/nushell/nushell/issues/5363
